code/modules/cards/cardhand.dm

```diff
@@ -0,0 +1,149 @@
+/obj/item/toy/cards/cardhand
+	name = "hand of cards"
+	desc = "A number of cards not in a deck, customarily held in ones hand."
+	icon = 'icons/obj/playing_cards.dmi'
+	icon_state = "none"
+	w_class = WEIGHT_CLASS_TINY
+	worn_icon_state = "card"
+
+/obj/item/toy/cards/cardhand/Initialize(mapload, list/cards_to_combine)
+	. = ..()
+
+	if(!LAZYLEN(cards_to_combine) && (mapload && !LAZYLEN(cards)))
+		CRASH("[src] is being made into a cardhand without a list of cards to combine")
+
+	if(mapload && LAZYLEN(cards)) // these cards have not been initialized
+		for(var/card_name in cards)
+			var/obj/item/toy/singlecard/new_card = new (loc, card_name)
+			new_card.update_appearance()
+			cards_to_combine += new_card
+		cards = list() // reset our cards to an empty list
+
+	if(LAZYLEN(cards_to_combine)) // these cards are already initialized
+		for(var/obj/item/toy/singlecard/new_card in cards_to_combine)
```

Comment:
you dont need to check the length of a list before iterating through it with a for loop, it wont iterate if theres nothing in the list

Reply:
Didn't know that.  Applied with https://github.com/tgstation/tgstation/pull/64200/commits/d768da604d5dc48f376a765a53a0313d65e7563f